templates/var/local/git/gitlab/config/unicorn.rb.j2

```diff
@@ -18,8 +18,10 @@
 # Read about unicorn workers here:
 # http://doc.gitlab.com/ee/install/requirements.html#unicorn-workers
 #
-{% if (ansible_local|d() and ansible_local.nginx|d() and ansible_local.nginx.flavor == 'passenger') %}
-worker_processes 1
+
+# Always have at least two worker processes (https://gitlab.com/gitlab-org/gitlab-ce/issues/13947)
+{% if (ansible_processor_cores == 1 or (ansible_local|d() and ansible_local.nginx|d() and ansible_local.nginx.flavor == 'passenger')) %}
+worker_processes 2
```

Comment:
That's even better. I'm not sure why the autoscaling was disabled with Passenger though.